### PR TITLE
L3AFd Docker build CI integration

### PR DESCRIPTION
**Pull request by aka320**
Building the Docker image and uploading it to Artifactory as part of CI build.

### CODE REVIEW

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -68,7 +68,20 @@ jobs:
...
+
       - uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808
         with:
           name: l3afd-latest-linux-x86_64
           path: l3afd
+
+      - uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808
+        with:
+          name: l3afd-docker-img-linux-x86_64-r2-latest
+          path: l3afd-docker-img-linux-x86_64-r2-latest.tar
```
Comment (sanfern) [inline .github/workflows/ci-build.yaml:87]:
```suggestion
          path: l3afd-docker-img-linux-x86_64-v2.0.0.tar
```

File: .github/workflows/ci-build.yaml [hunk 2]
```diff
@@ -68,7 +68,20 @@ jobs:
         run: |
           make
 
+      - name: Build docker image
+        run: |
+          sudo git clone https://github.com/l3af-project/l3af-arch.git /root/l3af-arch
+          sudo cp /root/l3af-arch/dev_environment/cfg/l3afd.cfg ./build-docker
+          sudo cp l3afd ./build-docker
+          sudo docker build -t l3afd:r2-latest -f ./build-docker/Dockerfile ./build-docker
+          sudo docker save l3afd:r2-latest > l3afd-docker-img-r2-latest.tar
```
Comment (sanfern) [inline .github/workflows/ci-build.yaml:77]:
```suggestion
          sudo docker save l3afd-docker-img-linux-x86_64-v2.0.0.tar
```

File: .github/workflows/ci-build.yaml [hunk 3]
```diff
@@ -68,7 +68,20 @@ jobs:
         run: |
           make
 
+      - name: Build docker image
+        run: |
+          sudo git clone https://github.com/l3af-project/l3af-arch.git /root/l3af-arch
+          sudo cp /root/l3af-arch/dev_environment/cfg/l3afd.cfg ./build-docker
```
Comment (sanfern) [inline .github/workflows/ci-build.yaml:74]: Avoid cloning of l3af-arch repo, its better to have local copy of l3afd.cfg file here.

**Review by sanfern — CHANGES_REQUESTED** [review]
Can we explore to upload the image to docker hub?

File: config/l3afd.cfg [hunk 4]
```diff
@@ -46,25 +45,28 @@ ingress-entry-function-name: tc_ingress_root
...
 [l3af-configs]
-restapi-addr: localhost:53000
+restapi-addr: 0.0.0.0:7080
 
 [l3af-config-store]
 filename: /var/l3afd/l3af-config.json
 
 [mtls]
-enabled: true
+enabled: false
```
Comment (pmoroney) [inline config/l3afd.cfg:58]: Is this file used for other use-cases, or as an example config for users? If so we might want to consider these changes and make sure this is what we want users to use by default.

File: config/l3afd.cfg [hunk 5]
```diff
@@ -27,8 +26,8 @@ n-metric-samples: 20
 [xdp-root]
 package-name: xdp-root
 artifact: l3af_xdp_root.tar.gz
-ingress-map-name: xdp_root_array
 command: xdp_root
+ingress-map-name: xdp_root_array
 version: latest
 object-file: xdp_root_kern.o
```
Comment (sanfern) [inline config/l3afd.cfg:32]:
```suggestion
object-file: xdp_root.bpf.o
```

**Review by sanfern — CHANGES_REQUESTED** [review]
Please update l3afd.cfg file to the latest, like tc_ingress_root.bpf.o and tc_egress_root.bpf.o

File: README.md [hunk 6]
```diff
@@ -47,7 +47,7 @@ cmake --build build
 - L3AFD binary & configuration that is required in the Docker image needs to be built locally and copied to build-docker directory
 - Execute below command to build the docker image
 ```
-docker build -t l3afd:r2 -f Dockerfile.l3afd .
+docker build -t l3afd:r2 -f Dockerfile .
```
Comment (sanfern) [inline README.md:50]:
```suggestion
docker build -t l3afd:v2.0.0 -f Dockerfile .
```
we need to change this version to v2.0.1 on the day of release.

**Review by sanfern — CHANGES_REQUESTED** [review]

**Comment by sanfern** [comment]
@aka320 Please rebase your branch with latest main.

**Pull request closed without merging**